ehrapy/anndata/anndata_ext.py

```diff
@@ -204,13 +212,17 @@ def move_to_x(adata: AnnData, to_x: list[str] | str) -> AnnData:
     """
     if isinstance(to_x, str):  # pragma: no cover
         to_x = [to_x]
-    new_adata = concat([adata, AnnData(adata.obs[to_x], dtype="object")], axis=1)
-    new_adata.obs = adata.obs[adata.obs.columns[~adata.obs.columns.isin(to_x)]]
-    # update uns (copy maybe: could be a costly operation but reduces reference cycles)
-    # users might save those as separate AnnData object and this could be unexpected behaviour if we dont copy
-    num_columns_moved, non_num_columns_moved, _ = _update_uns(adata, to_x, True)
-    new_adata.uns["numerical_columns"] = adata.uns["numerical_columns"] + num_columns_moved
-    new_adata.uns["non_numerical_columns"] = adata.uns["non_numerical_columns"] + non_num_columns_moved
+    if set(to_x).issubset(set(adata.var_names)):
```

Comment:
This only works while all items of `to_x` are in the `adata.var_names`. We need to also make this work in case only some of the columns in `to_x` are present in `adata.var_names` (and therefore copied before). So I guess we have to check for each column in `to_x` independently, whether it is in `var_names` or not.

And if not: they have to be added to `uns` as well (besides deleting them from `obs`) in the corresponding lists of numerical/non_numerical columns.

Would be cool, if you could add some tests as well for such cases.

Reply:
rest LGTM

Reply:
Thank you!